resen/Resen.py

```diff
@@ -895,13 +895,46 @@ def _get_home_dir(self):
         return os.path.join(homedir,appname)
 
 
+    def __process_exists(self,pid):
+        # Need to do different things for *nix vs. Windows
+        # see https://stackoverflow.com/questions/568271/how-to-check-if-there-exists-a-process-with-a-given-pid-in-python
+
+        if os.name == 'nt':
+            # only works on windows
+            from win32com.client import GetObject
+            WMI = GetObject('winmgmts:')
+            processes = WMI.InstancesOf('Win32_Process')
+            pids = [process.Properties_('ProcessID').Value for process in processes]
+            return pid in pids
+        else:
+            # only works on *nix systems
+            try:
+                os.kill(pid,0)
+            except ProcessLookupError:
+                return False
+            except PermissionError: # errno.EPERM
+                return True # Operation not permitted (i.e., process exists)
+            else:
+                return True # no error, we can send a signal to the process
+
+
     def __lock(self):
+        # dev note: if we want to be more advanced, need psutil as dependency
+        # get some telemetry to fingerprint with
+        cur_pid = os.getpid()      # process id
+
+        # check if lockfile exists
         self.__lockfile = os.path.join(self.resen_root_dir,'lock')
         if os.path.exists(self.__lockfile):
```

Comment:
With this new method of checking PIDs, is it necessary to remove the lock file at all after an instance of the Resen class is deleted, or would it be better to just leave it perpetually?  You still need some kind of logic here to generate the lock file the first time.

Reply:
If we don't remove the lockfile, there is a small chance that another process could spawn with the same PID as the previously crashed out process that imported Resen, so it is important to clean up the file to decrease the number of times this roll of the dice happens.

Other options are to add another dependency or to create a much more complicated lockfile system. With these, we might be able to get away with not cleaning up the file on exit.

Reply:
Yes, there is the chance that the terminal was closed, and resen crashed, and then there is a lock file with a PID saved in it. If resen is not used for some time, as @asreimer mentions,  there is a chance that another process could have spawn with that same PID.
1) A little addition to the lockfile system could be to create a random number/sequence (or hash) when starting resen and to accompany the PID
2) Actually I think the creation of a random sequence/number/hash when starting resen, and saved in the lockfile could be enough, no need for PID.
3) Another check could be asserting that the PID process is actually Python.

Reply:
Fair enough, sounds like cleaning up the lock file is the simplest solution and should work in 99% of cases.  We can keep some of these more complex options in mind if it becomes an issue in the future.